### PR TITLE
bpo-28494: Test existing zipfile working behavior.

### PR DESCRIPTION
**Pull request by gpshead**
Adds unittests for executables with a zipfile appended to test_zipfile
as zipfile.is_zipfile and zipfile.ZipFile work properly on these today.

This is being added as a regression test that https://github.com/python/cpython/pull/5053 would otherwise fail today.

https://bugs.python.org/issue28494

### CODE REVIEW

**Comment by gpshead** [comment]
duh, fixing the test to skip the execute one on windows :)

File: Lib/test/test_zipfile.py [hunk 1]
```diff
@@ -2496,11 +2496,15 @@ def test_read_zip64_with_exe_prepended(self):
         self._test_zip_works(self.exe_zip64)
 
     @unittest.skipUnless(sys.executable, 'sys.executable required.')
+    @unittest.skipUnless(os.access('/bin/bash', os.X_OK),
```
Comment (Yhg1s) [inline Lib/test/test_zipfile.py:2499]: How about using shutil.which('bash') instead? There are some windows machines with bash installed now... will the test work there?

Comment (gpshead) [reply Lib/test/test_zipfile.py:2499]: I didn't think windows supported #! line executables as a concept.  executable is based on filename ending in bat/cmd/com/exe, not an execute bit there.

**Comment by Yhg1s** [comment]
Buildbot failure should have been fixed by GH-15902.

**Pull request merged**